account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -76,7 +78,24 @@ def invoice2data_to_parsed_inv(self, invoice2data_res):
 
         for line in lines:
             # Manipulate line data to match with account_invoice_import
-            line['price_unit'] = float(line.get('price_unit', 0))
+            price_unit = line.get('price_unit', 0)
+
+            # Search currency information
+            currency = self.env['res.currency'].search([
+                ('name', '=', invoice2data_res.get('currency'))
```

Comment:
What if 'currency' does not occur in the invoice? Fallback to company currency?

Reply:
Added the fallback to company

Reply:
There's an active field `res.currency`, should we take that into account here?

Reply:
But `active` fields are always honoured implicitely?

Reply:
If the invoice currency is not active (and thus not found), should we then fall back to the company currency even if the currency actually exist?
I suppose my case is hypothetical though...

Reply:
Hypothetical indeed, because we don't really expect any Belgian Franc invoices at this point ;-)